src/dataconnect/schemaMigration.ts

```diff
@@ -43,60 +46,97 @@ export async function migrateSchema(args: {
   allowNonInteractiveMigration: boolean;
   validateOnly: boolean;
 }): Promise<Diff[]> {
-  const { schema, validateOnly } = args;
+  const { options, schema, allowNonInteractiveMigration, validateOnly } = args;
 
   const databaseId = schema.primaryDatasource.postgresql?.database;
   if (!databaseId) {
     throw new FirebaseError(
       "Schema is missing primaryDatasource.postgresql?.database, cannot migrate",
     );
   }
-  const instanceId = schema.primaryDatasource.postgresql?.cloudSql.instance.split("/").pop();
-  if (!instanceId) {
-    throw new FirebaseError(`tried to migrate schema but ${instanceId} was undefined`);
+  const instanceName = schema.primaryDatasource.postgresql?.cloudSql.instance;
+  if (!instanceName) {
+    throw new FirebaseError(
+      "tried to migrate schema but instance name was not provided in dataconnect.yaml",
+    );
   }
+  const instanceId = instanceName?.split("/").pop();
+
+  const serviceName = schema.name.replace(`/schemas/${SCHEMA_ID}`, "");
   try {
-    const serviceName = schema.name.replace(`/schemas/${SCHEMA_ID}`, "");
-    await ensureServiceIsConnectedToCloudSql(serviceName);
+    await ensureServiceIsConnectedToCloudSql(serviceName, instanceName, databaseId);
```

Comment:
Code health wise, I think we can extract a helper 

```
const (instanceName, databaseName) = await ensureServiceIsConnectedToCloudSql(schema);
```

This logic is shared between `sql:diff` and `sql:migrate`.

Do it outside `try catch`, don't think errors from this method should be handled in the same way.

Reply:
Good call - took a slightly different approach to have better separation of concerns, but still dryed this out